ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/blocks/BeaconBlockBody.java

```diff
@@ -212,4 +215,47 @@ public void setVoluntary_exits(List<VoluntaryExit> voluntary_exits) {
   public void setTransfers(List<Transfer> transfers) {
     this.transfers = transfers;
   }
+
+  public Bytes32 hash_tree_root() {
+    return HashTreeUtil.merkleize(
+        Arrays.asList(
+            HashTreeUtil.hash_tree_root_basic_type(randao_reveal.toBytes()),
```

Comment:
is this a basic type?

Reply:
So BLSSignature is one of the tricky ones, but at the end of the day it's serialized as 96-bytes without a length mixin, which makes it a tuple of basic types. Tuples of basic type follow the same rules as basic types themselves: `merkleize(pack(value)) if value is a basic object or a tuple of basic objects`. So in short: yes, I believe so.